Modules/Core/Common/include/itkThreadSupport.h

```diff
@@ -51,7 +51,7 @@ using FastMutexType = pthread_mutex_t;
 using ThreadFunctionType = void * (*)(void *);
 using ThreadProcessIdType = pthread_t;
 #  if !defined(ITK_FUTURE_LEGACY_REMOVE)
-constexpr ThreadProcessIdType ITK_DEFAULT_THREAD_ID = 0;
+constexpr ThreadProcessIdType ITK_DEFAULT_THREAD_ID = nullptr;
```

Comment:
I am uncomfortable with this change. Where does the warning appear?

Reply:
The warning appears on MacOS. pthread_t is a pointer. However, on Linux, it is an int. We will remove the change.

Reply:
@andrei-sandor @dzenanz If the warning still bothers you, I would suggest using `{}` instead. `{}` should be universal, so it should work well for both pointers and integers:

    constexpr ThreadProcessIdType ITK_DEFAULT_THREAD_ID{};